app/models/application.rb

```diff
@@ -0,0 +1,25 @@
+class Application < ApplicationRecord
+  validates :applicant_name, presence: true
+  validates :address,        presence: true
+  validates :city,           presence: true
+  validates :state,          presence: true
+  validates :zip_code,       presence: true
+  validates :description,    presence: true
+
+  has_many :pet_applications
+  has_many :pets, through: :pet_applications
+  
+  def search_pets_by_name(name)
+    Pet.search_by_name(name)
```

Comment:
Check out the ApplicationRecord -- there is a search function already built for you there that you can use! It is inherited into all models because all models inherit from ApplicationRecord. That means, you could call `Pet.search(search_term)` without implementing search again. 

Also, this method doesn't include querying the application table at all so it would make more sense in the Pet model if it were not already available